### DISCUSSION TITLE
Any way to further distinguish between PowerTag devices?

### DISCUSSION THREAD

**Discussion by tkarlsson** [Koenkk/zigbee2mqtt#18625] [Devices]
I'm keen to have further granularity than a fingerprint based on ieee for Schneider PowerTags. Reason being that there are different models with different capabilities and distinguishing between the different models would help create the right set of sensors.

For instance, in the commandNotification this information is available which idicates the right model number:

-------
Zigbee2MQTT:debug 2023-01-13 23:59:24: Received Zigbee message from '0x00000000e203e53f', type 'commandNotification', cluster 'greenPower', data '{"commandFrame":{"attributes":{"57352":"Resi9 PowerTag","57353":"R9 M63 3P"},"clusterID":0,"manufacturerCode":4190},"commandID":161,"frameCounter":20,"gppGddLink":255,"gppNwkAddr":39135,"options":22544,"payloadSize":35,"srcID":3791906111}' from endpoint 242 with groupID 58687
-------

Anyone that has any recommendations?

**Comment by KimCarlsen** [comment]
hi
did you ever found a solution?
I have an R9M20 and a R9M41 powertag and both shows up as a A9MEM1570. one of the is a single phase and therefor pretty off. the other shows negative values,.

**Comment by KimCarlsen** [comment]
my generated_external definitions:
`
//R9M20 1P

import {genericGreenPower} from 'zigbee-herdsman-converters/lib/modernExtend';

export default {
    fingerprint: [{modelID: 'GreenPower_254', ieeeAddr: new RegExp('^0x00000000e205e711$')}],
    model: 'GreenPower_254',
    vendor: '',
    description: 'Automatically generated definition for Green Power',
    extend: [genericGreenPower()],
};

//R9M41 3P+n
import {genericGreenPower} from 'zigbee-herdsman-converters/lib/modernExtend';

export default {
    fingerprint: [{modelID: 'GreenPower_254', ieeeAddr: new RegExp('^0x00000000e2067e3f$')}],
    model: 'GreenPower_254',
    vendor: '',
    description: 'Automatically generated definition for Green Power',
    extend: [genericGreenPower()],
};`

**Reply by KCarlsen** [reply]
and my second R9M20.
import {genericGreenPower} from 'zigbee-herdsman-converters/lib/modernExtend';

export default {
    fingerprint: [{modelID: 'GreenPower_254', ieeeAddr: new RegExp('^0x00000000e204afd7$')}],
    model: 'GreenPower_254',
    vendor: '',
    description: 'Automatically generated definition for Green Power',
    extend: [genericGreenPower()],
};

So seems the current external definition is not enough to identify them
